### PR TITLE
update Meta::thriftIDL

### PR DESCRIPTION
**Pull request by ShanniLi**
r @Raynos @kriskowal 
cc @junchaowu @prashantv @blampe

### CODE REVIEW

File: as/meta.thrift [hunk 1]
```diff
@@ -3,7 +3,14 @@ struct HealthStatus {
     2: optional string message
 }
 
+struct ThriftIDLs {
+    // map: filename -> contents
+    1: required map<string, string> idls
+    // the entry IDL that imports others
+    2: required string entryPoint
```
Comment (kriskowal) [inline as/meta.thrift]: consider `typedef filename string` and using the filename type here to disambiguate the IDL for the humans.

**Comment by kriskowal** [comment]
lgtm

File: as/thrift.js [hunk 2]
```diff
@@ -45,6 +45,7 @@ function TChannelAsThrift(opts) {
         'must pass source as an argument');
 
     self.thriftSource = opts.source;
+    self.thriftFileName = opts.thriftFileName || 'service';
```
Comment (kriskowal) [inline as/thrift.js]: consider `service.thrift` as the default.

File: as/meta.thrift (hunk 1 above)
Comment (ShanniLi) [inline as/meta.thrift]: Good point. Need to land this first:
https://github.com/uber/tchannel/pull/1390

File: as/thrift.js (hunk 2 above)
Comment (ShanniLi) [inline as/thrift.js]: Not sure like the idea of having a dot in the map key. Plus, it should always be ".thrift'. Will make the change if got a '+1' on 'service.thrift'. :)

Comment (ShanniLi) [inline as/thrift.js]: Actually, I am convinced. Will make the change.

File: as/thrift.js [hunk 3]
```diff
@@ -45,6 +45,7 @@ function TChannelAsThrift(opts) {
         'must pass source as an argument');
 
     self.thriftSource = opts.source;
+    self.thriftFileName = opts.thriftFileName || 'service.thrift';
```
Comment (Raynos) [inline as/thrift.js]: nope.

This argument must either be undefined or not.

We should only implement `thriftIDL` if this is specified.

Comment (Raynos) [inline as/thrift.js]: Also please update https://github.com/uber/tchannel-node/blob/master/hyperbahn-client.js#L179-L197 to pass argument.

Comment (kriskowal) [inline as/thrift.js]: We can follow-up to support AsThrift({idls, entryPoint}) when ThriftRW supports that form of constructor. For now, this is a fine migration step.

Comment (ShanniLi) [inline as/thrift.js]: oops, click too fast ... Will create a separate PR ...

**Pull request merged**